### PR TITLE
Add sanity check for date type for sequence function

### PR DESCRIPTION
**Pull request by kewang1024**

### CODE REVIEW

File: velox/functions/prestosql/tests/SequenceTest.cpp [hunk 1]
```diff
@@ -207,6 +207,18 @@ TEST_F(SequenceTest, dateInvalidIntervalDayStep) {
       expected);
 }
 
+TEST_F(SequenceTest, invalidDate) {
+  int64_t day = 86400000; // 24 * 60 * 60 * 1000
+  const auto startVector = makeFlatVector<int32_t>({1420919120, 1992}, DATE());
+  const auto stopVector = makeFlatVector<int32_t>({1476733096, 2000}, DATE());
+  auto stepVector =
+      makeFlatVector<int64_t>({1 * day, 0, 1}, INTERVAL_YEAR_MONTH());
```
Comment (gggrace14) [inline velox/functions/prestosql/tests/SequenceTest.cpp:215]: stepVector has different size than the startVector and stopVector? Probably vectors of size 1 is sufficient for this test.

File: velox/functions/prestosql/Sequence.cpp [hunk 2]
```diff
@@ -156,13 +156,27 @@ class SequenceFunction : public exec::VectorFunction {
   }
 
  private:
+  static void isDateValid(const int32_t date) {
+    // Find the number of seconds for the date;
+    // Casting 86400 to int64 to handle overflows gracefully.
+    int64_t daySeconds = date * (int64_t)(86400);
+    auto tmValue = gmtime((const time_t*)&daySeconds);
+    VELOX_USER_CHECK(
+        tmValue->tm_year > SHRT_MAX || tmValue->tm_year < 0,
```
Comment (gggrace14) [inline velox/functions/prestosql/Sequence.cpp:165]: You might know it, VELOX_USER_CHECK() is to check if the first arg is true. If it is not, throw the error exception.

File: velox/functions/prestosql/Sequence.cpp [hunk 3]
```diff
@@ -156,13 +156,27 @@ class SequenceFunction : public exec::VectorFunction {
   }
 
  private:
+  static void isDateValid(const int32_t date) {
+    // Find the number of seconds for the date;
+    // Casting 86400 to int64 to handle overflows gracefully.
+    int64_t daySeconds = date * (int64_t)(86400);
+    auto tmValue = gmtime((const time_t*)&daySeconds);
+    VELOX_USER_CHECK(
+        tmValue->tm_year < SHRT_MAX && tmValue->tm_year > 0,
```
Comment (gggrace14) [inline velox/functions/prestosql/Sequence.cpp:165]: I guess tmValue->tm_year >= 0, to be precise.

**Review by gggrace14 — APPROVED** [review]
One minor change requested. Otherwise, approving...

File: velox/functions/prestosql/Sequence.cpp [hunk 4]
```diff
@@ -156,13 +156,27 @@ class SequenceFunction : public exec::VectorFunction {
   }
 
  private:
+  static void isDateValid(const int32_t date) {
+    // Find the number of seconds for the date;
+    // Casting 86400 to int64 to handle overflows gracefully.
+    int64_t daySeconds = date * (int64_t)(86400);
+    auto tmValue = gmtime((const time_t*)&daySeconds);
+    VELOX_USER_CHECK(
```
Comment (mbasmanova) [inline velox/functions/prestosql/Sequence.cpp:164]: Usse 2 separate checks for better error message. VELOX_USER_CHECK_LT and ..._GT

**Review by gggrace14 — COMMENTED** [review]
Oh, also want to check with @laithsakka , if we throw exception in UDF, what will happen to Fuzzer? Will it resolve the Fuzzer failure?

**Review by mbasmanova — COMMENTED** [review]
@kewang1024 Ke, would you update PR description to explain the problem and proposed solution. Is this behavior consistent with Presto?

**Comment by laithsakka** [comment]
thanks for working on this 
but it looks like this does not match presto semantics

when i try this on presto it succeed, but it sound like it wont succeed here
(1) :
```select SEQUENCE(cast ('3892314-06-02'as DATE), cast ('3892314-06-02'as DATE))```

the output is 
[
  "+3892314-06-02"
]

also when run 
(2):
```
select SEQUENCE(cast ('3892314-06-02'as DATE), cast ('3892329-06-02'as DATE))
```
```

  "+3892314-06-02",
  "+3892314-06-03",
  "+3892314-06-04",
  "+3892314-06-05",
  "+3892314-06-06",
  "+3892314-06-07" ...
  ```
for 

(3):
```select SEQUENCE(cast ('3892314-06-02'as DATE), cast('4045127-11-23'as DATE))```
the expected error message is 
```result of sequence function must not have more than 10000 entries```


also can you add a summary and refer to the fuzzer issue.

**Comment by gggrace14** [comment]
> @kewang1024 Ke, would you update PR description to explain the problem and proposed solution. Is this behavior consistent with Presto?

@mbasmanova I will be looking at the behavior of date_diff today.

**Comment by kewang1024** [comment]
Thanks Laith for catching the issue, if we match java presto’s behavior, then we need to adjust the utilities function (diffTimestamp) instead of sequence function.
On sequence function level, it can only limit the data, however java presto doesn’t have that limit.

After discussing offline, we decide to fix the date_diff instead.

**Pull request closed without merging**